ts/Series/CompareCumulativeComposition.ts

```diff
@@ -0,0 +1,789 @@
+     * @private
+     */
+    function afterGetExtremes(this: Series, e: AnyRecord|Event): void {
+        const dataExtremes: DataExtremesObject = (e as any).dataExtremes,
+            activeYData = dataExtremes.activeYData;
+
+        if (this.dataModify && this.dataModify.modifyValue && dataExtremes) {
+            let extremes;
+
+            if (this.options.compare) {
+                extremes = [
+                    this.dataModify.modifyValue(dataExtremes.dataMin),
+                    this.dataModify.modifyValue(dataExtremes.dataMax)
+                ];
+            } else if (
+                this.options.cumulative &&
+                isArray(activeYData) &&
+                // If only one y visible, sum doesn't change
+                // so no need to change extremes
+                activeYData.length >= 2
+            ) {
+                extremes = Additions.getCumulativeExtremes(activeYData);
+            }
+
+            if (extremes) {
+                dataExtremes.dataMin = arrayMin(extremes);
+                dataExtremes.dataMax = arrayMax(extremes);
+            }
+        }
+    }
+
+    /* ********************************************************************** *
+     *  End shared compare and cumulative logic                               *
+     * ********************************************************************** */
+
+    /* ********************************************************************** *
+     *  Start value compare logic                                             *
+     * ********************************************************************** */
+
+    /**
+     * Highcharts Stock only. Set the
+     * [compare](https://api.highcharts.com/highstock/plotOptions.series.compare)
+     * mode of the series after render time.
+     * In most cases it is more useful running
+     * {@link Axis#setCompare} on the X axis to update all its series.
+     *
+     * @function Highcharts.Series#setCompare
+     *
+     * @param {string|null} [compare]
+     *        Can be one of `undefined` (default), `null`, `"percent"`
+     *        or `"value"`.
+     *
+     * @param {boolean} [redraw=true]
+     *        Whether to redraw the chart or to wait for a later call to
+     *        {@link Chart#redraw}.
+     */
+    function seriesSetCompare(
+        this: Series,
+        compare?: string|null,
+        redraw?: boolean
+    ): void {
+        this.dataModify && this.dataModify.initCompare(compare);
+
+        // Survive to export, #5485
+        this.options.compare = this.userOptions.compare = compare;
+
+        if (pick(redraw, true)) {
+            this.chart.redraw();
+        }
+    }
+
+    /**
+     * Extend series.processData by finding the first y value in the plot area,
+     * used for comparing the following values
+     *
+     * @ignore
+     * @function Highcharts.Series#processData
+     */
+    function afterProcessData(this: Series): (boolean|undefined) {
```

Comment:
I'm not sure if this function returns any of that.

Reply:
Copied from the old `compare`. Corrected